File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    match = {')': '('}
    stack = []
    for c in s:
        if c in '(':
            stack.append(c)
        elif c in match:
            if len(stack) == 0:
                return False
            else:
                t = stack.pop()
                if t != match[c]:
                    return False
    answer = True if not bool(len(stack)) else False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Integer> sa = new Stack<>();
        for(int i = 0; i<s.length(); i++){
            char num = s.charAt(i);
            if(num == '('){
                sa.add(1);
            }else{
                if(sa.isEmpty()){
                    answer = false;
                    break;
                }
                sa.pop();
            }
        }
        
        answer = !sa.isEmpty() ? false : answer;
        return answer;
    }
}